AMM/project/contracts/AMM-contract/tests/functions/initialize.rs

```diff
@@ -0,0 +1,42 @@
+use crate::utils::{
+    amm_abi_calls::initialize,
+    test_helpers::{
+        bytecode_root, deploy_and_construct_exchange_contract, setup, setup_and_initialize,
+    },
+    HARDCODED_EXCHANGE_CONTRACT_BYTECODE_ROOT,
+};
+use fuels::tx::ContractId;
+use std::str::FromStr;
+
+mod success {
+    use super::*;
+
+    #[tokio::test]
+    async fn initializes() {
+        let (_wallet, amm_instance, _asset_pairs) = setup().await;
+
+        let hardcoded_bytecode_root =
+            ContractId::from_str(HARDCODED_EXCHANGE_CONTRACT_BYTECODE_ROOT).unwrap();
+        let calculated_bytecode_root = bytecode_root().await;
+
+        initialize(&amm_instance, hardcoded_bytecode_root).await;
+
+        assert_eq!(calculated_bytecode_root, hardcoded_bytecode_root);
```

Comment:
I don't think we need the hardcoded root if we just compare it to the calculated one and assert they are equal. We can just use the calculated one directly: 

```suggestion
        let calculated_bytecode_root = bytecode_root().await;
        initialize(&amm_instance, calculated_bytecode_root ).await;
```